src/middlewared/middlewared/plugins/disk.py

```diff
@@ -1798,16 +1798,18 @@ def format(self, disk, swapgb, sync=True):
             # The GPT header takes about 34KB + alignment, round it to 100
             if size - 100 <= swapgb * 1024 * 1024:
                 raise CallError(f'Your disk size must be higher than {swapgb}GB')
+            sectorsize = g.provider.sectorsize
```

Comment:
Is there any chance this could be null or empty? Would it be safer to put `or 512`?

Also is this not needed for 12.1?

Reply:
@themylogin ping

Reply:
@william-gr I'll backport from 12.0 to 12.1